apps/aecore/src/aec_vm_chain.erl

```diff
@@ -459,6 +463,44 @@ oracle_response_format(Oracle, #state{trees = ChainTrees}) ->
             {error, no_such_oracle}
     end.
 
+oracle_check(Oracle, QFormat, RFormat, #state{trees = ChainTrees}) ->
+    Trees = get_on_chain_trees(ChainTrees),
+    case aeo_state_tree:lookup_oracle(Oracle, aec_trees:oracles(Trees)) of
+        {value, O} ->
+            {ok, ChainRFormat} = oracle_typerep(aeo_oracles:abi_version(O),
+                                                aeo_oracles:response_format(O)),
+            {ok, ChainQFormat} = oracle_typerep(aeo_oracles:abi_version(O),
+                                                aeo_oracles:query_format(O)),
+            Equal = (QFormat == ChainQFormat) andalso (RFormat == ChainRFormat),
+            {ok, bool2word(Equal)};
+        none ->
+            {ok, bool2word(false)}
+    end.
+
+oracle_check_query(Oracle, Query, QFormat, RFormat, #state{trees = ChainTrees}) ->
+    Trees = get_on_chain_trees(ChainTrees),
+    OraclesTree = aec_trees:oracles(Trees),
+    case aeo_state_tree:lookup_query(Oracle, Query, OraclesTree) of
+        {value, _} ->
+            {value, O} = aeo_state_tree:lookup_oracle(Oracle, OraclesTree),
+            {ok, ChainRFormat} = oracle_typerep(aeo_oracles:abi_version(O),
+                                                aeo_oracles:response_format(O)),
+            {ok, ChainQFormat} = oracle_typerep(aeo_oracles:abi_version(O),
+                                                aeo_oracles:query_format(O)),
+            ABIVersion         = aeo_oracles:abi_version(O),
+            case ABIVersion of
+                ?ABI_NO_VM -> %% Question/Response is non-sophia string
+                    Equal = string == QFormat andalso string == ChainQFormat andalso
+                            string == RFormat andalso string == ChainRFormat,
+                    {ok, bool2word(Equal)};
+                _Other ->
+                    Equal = (QFormat == ChainQFormat) andalso (RFormat == ChainRFormat),
+                    {ok, bool2word(Equal)}
+            end;
```

Comment:
`ABIVersion` check should not be necessary, and the whole block can be replace with a call to `oracle_check`.